fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -534,8 +564,10 @@ private SqlQueryData processReferenceParm(Class<?> resourceType, QueryParameter
                                     + SqlParameterEncoder.encode(value.getValueString());
                 } else {
                     // This is a Reference type.
-                    // As of versions greater than 4.4.0, we defer to the Search Layer to append the value.
-                    searchValue = SqlParameterEncoder.encode(value.getValueString());
+                    if (parts.length != 2) {
```

Comment:
Could be combined to one else if

Reply:
Could be, but I prefer not to because the predicate is based on a different variables than the enclosing if. It also helps to keep the Not a Reference/Reference comments neater.